compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -245,13 +250,45 @@ impl<K: DepKind> DepGraph<K> {
     /// in the query infrastructure, and is not currently needed by the
     /// decoding of any query results. Should the need arise in the future,
     /// we should consider extending the query system with this functionality.
-    pub fn with_query_deserialization<OP, R>(&self, op: OP) -> R
+    pub fn with_query_deserialization<OP, R>(op: OP) -> R
```

Comment:
Why not leave it a method?

Reply:
I just removed the uses of `dep_graph`. I could probably keep this and check that LLVM does manage to remove the argument.